custom/icds_reports/utils/aggregation_helpers/monolith/daily_feeding_forms_child_health.py

```diff
@@ -38,19 +38,22 @@ def aggregation_query(self):
           state_id, supervisor_id, month, case_id, latest_time_end_processed,
           sum_attended_child_ids, lunch_count
         ) (
-          SELECT DISTINCT ON (child_health_case_id)
+          SELECT DISTINCT ON (ucr.child_health_case_id)
             %(state_id)s AS state_id,
-            LAST_VALUE(supervisor_id) OVER w AS supervisor_id,
+            LAST_VALUE(ucr.supervisor_id) OVER w AS supervisor_id,
             %(month)s AS month,
-            child_health_case_id AS case_id,
-            MAX(timeend) OVER w AS latest_time_end_processed,
-            SUM(attended_child_ids) OVER w AS sum_attended_child_ids,
-            SUM(lunch) OVER w AS lunch_count
-          FROM "{ucr_tablename}"
-          WHERE state_id = %(state_id)s AND
-                timeend >= %(current_month_start)s AND timeend < %(next_month_start)s AND
-                child_health_case_id IS NOT NULL
-          WINDOW w AS (PARTITION BY child_health_case_id)
+            ucr.child_health_case_id AS case_id,
+            MAX(ucr.timeend) OVER w AS latest_time_end_processed,
+            SUM(ucr.attended_child_ids) OVER w AS sum_attended_child_ids,
+            SUM(ucr.lunch) OVER w AS lunch_count
+          FROM "{ucr_tablename}" ucr inner join daily_attendance ON (
+            ucr.doc_id = daily_attendance.doc_id AND
+            daily_attendance.month=%(current_month_start)s
```

Comment:
what's the purpose of joining here? this table doesn't get used anywhere else in the query

Reply:
Hey The purpose is just to limit the number of rows we pick up from ucr.
Ideally we want to pick rows whose corresponding doc_id is present in daily_attendance. 
Purpose of join was not to use any property from that table.